### PR TITLE
ban in redis cache

### PR DESCRIPTION
**Pull request by khodzha**

### CODE REVIEW

**Review by feymartynov — APPROVED** [review]

**Review by manifest — APPROVED** [review]
`ban_in_cache` -> `ban`

**Review by feymartynov — APPROVED** [review]

File: src/intent.rs [hunk 1]
```diff
@@ -3,15 +3,33 @@ use svc_authn::AccountId;
 
 ////////////////////////////////////////////////////////////////////////////////
 
+pub trait Object: Send + Sync {
+    fn ban_key(&self) -> Option<String>;
+    fn as_vec(&self) -> Vec<String>;
```
Comment (manifest) [inline src/intent.rs:8]: Wouldn't it be more consistent to have:
```rust
fn to_ban_key(&self) -> Option<Vec<String>>;
fn to_key(&self) -> Vec<String>;
```

So, for both we would later use
```
obj.to_ban_key().join("/")
obj.to_key().join("/")
```

File: src/cache.rs [hunk 2]
```diff
@@ -21,26 +21,72 @@ pub fn create_pool(url: &str, size: u32, idle_size: Option<u32>, timeout: u64) -
...
+            let result: Result<bool, _> = conn.set_ex(key, value as u8, expiration_seconds);
+            if Ok(true) == result {
+                return;
+            }
+        }
+
+        error!("Cache is unavailable");
+    }
+
+    fn get_keys(&self, keys: &[&str]) -> Vec<Response> {
```
Comment (manifest) [inline src/cache.rs:70]: Just `keys`?

File: src/lib.rs [hunk 3]
```diff
@@ -306,69 +369,114 @@ impl IntoClient for HttpConfig {
...
+                            Ok(())
+                        } else {
+                            Err(ErrorKind::Forbidden(IntentError::new(
+                                intent,
+                                "the action forbidden by tenant (cache hit)",
+                            ))
+                            .into())
+                        };
+                    }
+                    [_, _] => {}
```
Comment (manifest) [inline src/lib.rs:453]: `Ok(())`?

File: src/lib.rs [hunk 4]
```diff
@@ -306,69 +369,114 @@ impl IntoClient for HttpConfig {
...
+                    _ => {
+                        log::warn!("Cache ban request returned improper number of responses, expected 2, got = {:?}", cache_response);
```
Comment (manifest) [inline src/lib.rs:455]: `Ok(())` ?

File: src/cache.rs (hunk 2 above)
Comment (khodzha) [reply src/cache.rs:70]: i dont like `keys` because it collides with actual `KEYS` redis command

File: src/lib.rs (hunk 3 above)
Comment (khodzha) [reply src/lib.rs:455]: this is fallthrough case, we dont early return, no reason to use `Ok(())` here

File: src/lib.rs (hunk 4 above)
Comment (khodzha) [reply src/lib.rs:457]: same as above

**Review by manifest — APPROVED** [review]

**Pull request merged**